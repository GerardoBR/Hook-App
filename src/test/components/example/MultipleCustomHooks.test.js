import { shallow } from "enzyme"
import { MultipleCustomHook } from "../../../components/example/MultipleCustomHook"
import { useCounter } from "../../../hooks/useCounter";
import { useFetch } from "../../../hooks/useFetch";
jest.mock("../../../hooks/useFetch");
jest.mock("../../../hooks/useCounter");
describe('Pruebas en Multimple custom hooks', () => {

    
    beforeEach( () => {
        useCounter.mockReturnValue({
            counter: 10,
            increment: () => {}
        })
    })



    test('debe mostrarse correctamente el componente ', () => {
        useFetch.mockReturnValue({
            data : null,
            loading : true,
            error : null   
        })
        const wrapper = shallow (<MultipleCustomHook/>);
        expect( wrapper ).toMatchSnapshot();

    })
    
    test('debe de mostrar la informacion', () => {

        useFetch.mockReturnValue({
            data : [{
                author: 'Gerardo',
                quote : 'El hambre es cabrona'
            }],
            loading : false,
            error : null   
        })
        const wrapper = shallow (<MultipleCustomHook/>);
        expect( wrapper.find('.alert').exists()).toBe(false);
        expect( wrapper.find('p').text().trim()).toBe('El hambre es cabrona');
        expect( wrapper.find('figcaption').text().trim()).toBe('Gerardo');
        // console.log(wrapper.find('p').text().trim())
        // console.log( wrapper.html());
    })
    
})
