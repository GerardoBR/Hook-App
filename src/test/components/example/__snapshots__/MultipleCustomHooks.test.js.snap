// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Multimple custom hooks debe mostrarse correctamente el componente  1`] = `
<div>
  <h1>
    Breaking Bad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-primary text-center"
    role="alert"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
  >
     Siguiente frase
  </button>
</div>
`;
